pyfr/plugins/tavg.py

```diff
@@ -358,3 +363,248 @@ def __call__(self, intg):
                     self.tstart_acc = intg.tcurr
 
                 self.tout_last = intg.tcurr
+
+
+class TavgCLIPlugin(BaseCLIPlugin):
+    name = 'tavg'
+
+    @classmethod
+    def add_cli(cls, parser):
+        cmd = cls.name
+        cp = parser.add_parser(cmd, help=f'{cmd} --help')
+        sp = cp.add_subparsers(dest='sub_cmd')
+
+        # Merge command
+        ssp = sp.add_parser('merge', help=f'{cmd} merge --help')
+        ssp.set_defaults(process=cls.merge_cli)
+        ssp.add_argument('solns', nargs='*', help='Averages to merge')
+        ssp.add_argument('output', help='Output file name')
+
+    @cli_external
+    def merge_cli(self):
+        # Open all the solution files
+        self._prepare_files(self.args.solns)
+
+        # Initialise things needed for the merge
+        self._init_tavg_merge()
+
+        with h5py.File(self.args.output, 'w') as outf:
+            # Merge the averages and stds
+            self._merge_data(outf)
+
+            # Merge the metadata
+            self._merge_meta(outf)
+
+    def _eval_fun_exprs(self, avars):
+        # Prepare the substitution dictionary
+        subs = dict(zip(self.anames, avars.swapaxes(0, 1)))
+
+        # Evaluate the function and return
+        return np.stack([npeval(v, subs) for v in self.fexprs], axis=1)
+
+    def _eval_fun_var(self, std, avars):
```

Comment:
There's a lot of code duplication here with the main plugin.  Worth refactoring to either a mixin or standalone function.

Reply:
Both version of this function do slightly different things. The main plugin is given region information and then calculates the fun var, whereas in the CLI version doens't have access region information and also works on a tranposed version of the data.

Reply:
I think this body:
```
            # Evaluate the function
            fx = self._eval_fun_exprs(av)
            exprs.append(fx)

            for avi in av:
                # Calculate step size for finite difference
                h = self.delta_h*np.abs(avi)
                h[np.where(h == 0)] = self.delta_h

                # Calculate derivatives for functional averages
                avi += h
                df.append((self._eval_fun_exprs(av) - fx) / h)
                avi -= h
```
Can still be reasonably factored out into a function, even if it means one of the callers having to do a bit of pre/post transposing.